cheatsheets/Browser_Extension_Vulnerabilities_Cheat_Sheet.md

```diff
@@ -1,43 +1,217 @@
+🚨 Vulnerability
+Some extensions unintentionally expose user data by sending browsing activity or personal details to external servers without proper security measures.
 
-## 5. Code Injection
+🔍 Example
 
-If an attacker controls the script URL, they can inject malicious code into the page, leading to data theft or manipulation of the page’s functionality.
+javascript
+Copy code
+chrome.tabs.onUpdated.addListener((tabId, changeInfo, tab) => {
+  if (changeInfo.status === 'complete') {
+    fetch('http://example.com/track', {
+      method: 'POST',
+      body: JSON.stringify({ url: tab.url })
+    });
+  }
+});
+🔒 Mitigation
 
-## 6. Malicious Updates
+Always use HTTPS for all communications to prevent data interception.
+Limit data collection and be transparent by clearly stating what data is collected in a Privacy Policy.
+Implement user consent mechanisms before collecting or sending any personal data.
 
-If the update mechanism is compromised, attackers can push malicious code to users without their knowledge, potentially gaining control over their browsers.
 
-## 7. Third-Party Dependencies
+3. Cross-Site Scripting (XSS)
 
-An extension relying on outdated third-party libraries may become vulnerable if those libraries have known security flaws that attackers can exploit.
+🚨 Vulnerability
+If user input is not properly sanitized, attackers can inject malicious scripts into web pages, potentially stealing user data or performing unauthorized actions.
 
-## 8. Lack of Content Security Policy (CSP)
+🔍 Example
 
-Without a strong CSP, attackers can inject untrusted content, increasing the risk of XSS and other attacks that manipulate the extension’s behavior.
+javascript
+Copy code
+let userInput = document.getElementById('input').value;
+document.getElementById('output').innerHTML = userInput; // No sanitization
+🔒 Mitigation
 
-## 9. Insecure Storage
+Implement Content Security Policy (CSP) to block inline scripts.
+Use libraries like DOMPurify to sanitize user input before displaying it.
+Avoid using innerHTML and instead use textContent to prevent execution of injected scripts.
 
-If an attacker gains access to the local storage, they can easily retrieve sensitive information, such as tokens or user credentials, leading to unauthorized access.
 
-## 10. Insufficient Privacy Controls
+4. Insecure Communication
 
-Users may be unaware of how their data is being collected or used, leading to potential abuse of their information without consent or awareness.
+🚨 Vulnerability
+Some extensions send sensitive data over unsecured HTTP connections, making it vulnerable to interception by attackers.
+
+🔍 Example
+
+javascript
+Copy code
+fetch('http://example.com/api/data');
+🔒 Mitigation
+
+Always use HTTPS for external communications to prevent data theft.
+Where possible, implement certificate pinning to prevent man-in-the-middle (MITM) attacks.
```

Comment:
I'd recommend removing this. See https://cheatsheetseries.owasp.org/cheatsheets/Pinning_Cheat_Sheet.html#when-do-you-perform-pinning

```suggestion
```